toqito/channels/tests/test_generalized_amplitude_damping.py

```diff
@@ -0,0 +1,113 @@
+"""Tests for generalized amplitude damping channel."""
+
+import re
+
+import numpy as np
+import pytest
+
+from toqito.channels import generalized_amplitude_damping
+
+
+@pytest.mark.parametrize("prob, gamma", [(0.0, 0.0), (0.3, 0.5), (0.5, 0.7), (1.0, 1.0), (0.0, 1.0), (1.0, 0.0)])
+def test_kraus_operators(prob, gamma):
+    """Test if the function returns correct Kraus operators for given probability and gamma."""
+    kraus_ops = generalized_amplitude_damping(None, prob=prob, gamma=gamma)
+
+    k0_expected = np.sqrt(prob) * np.array([[1, 0], [0, np.sqrt(1 - gamma)]])
+    k1_expected = np.sqrt(prob) * np.sqrt(gamma) * np.array([[0, 1], [0, 0]])
+    k2_expected = np.sqrt(1 - prob) * np.array([[np.sqrt(1 - gamma), 0], [0, 1]])
+    k3_expected = np.sqrt(1 - prob) * np.sqrt(gamma) * np.array([[0, 0], [1, 0]])
+    expected_kraus_ops = [k0_expected, k1_expected, k2_expected, k3_expected]
+
+    for i in range(4):
+        np.testing.assert_almost_equal(kraus_ops[i], expected_kraus_ops[i])
+
+    completeness_sum = np.zeros((2, 2), dtype=complex)
+    for k in kraus_ops:
+        completeness_sum += k.conj().T @ k
+    np.testing.assert_almost_equal(completeness_sum, np.eye(2))
+
+
+@pytest.mark.parametrize(
+    "rho, prob, gamma",
+    [
+        # Ground state |0⟩⟨0|.
+        (np.array([[1, 0], [0, 0]]), 0.3, 0.4),
+        # Excited state |1⟩⟨1|.
+        (np.array([[0, 0], [0, 1]]), 0.3, 0.4),
+        # Superposition state (|0⟩+|1⟩)(⟨0|+⟨1|)/2.
+        (np.array([[0.5, 0.5], [0.5, 0.5]]), 0.3, 0.4),
+        # Mixed state.
+        (np.array([[0.7, 0.2j], [-0.2j, 0.3]]), 0.3, 0.4),
+        # Complex mixed state.
+        (np.array([[0.7, 0.2 + 0.1j], [0.2 - 0.1j, 0.3]], dtype=complex), 0.4, 0.6),
+    ],
+)
+def test_apply_to_states(rho, prob, gamma):
+    """Apply the channel to various input states rho."""
+    kraus_ops = generalized_amplitude_damping(None, prob=prob, gamma=gamma)
+
+    expected_output = np.zeros((2, 2), dtype=complex)
+    for k in kraus_ops:

```

Comment:
Could these two tests also be combined where the first input argument is `None` for the Kraus operator case and where it is the density matrix in the other case?